From ea1b315c293084bdc268992a5bc49320b97fc99e Mon Sep 17 00:00:00 2001
From: Arulanbu Balusamy <quic_abalusam@quicinc.com>
Date: Tue, 14 Nov 2023 15:44:04 +0530
Subject: [PATCH] wifi: ath12k: Modify the power save flag of the Green AP

Modify the power save flag in ath12k_mac_bss_info_changed from BSS_CHANGED_PS
to BSS_CHANGED_AP_PS because in the mac80211 configure the seperate
flag for AP and STA.

New configuration flags:
========================
BSS_CHANGED_PS for STA
BSS_CHANGED_AP_PS for AP

Signed-off-by: Arulanbu Balusamy <quic_abalusam@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -4841,7 +4841,7 @@
 	    changed & BSS_CHANGED_UNSOL_BCAST_PROBE_RESP)
 		ath12k_mac_fils_discovery(arvif, info);
 
-	if ((changed & BSS_CHANGED_PS) && vif->type == NL80211_IFTYPE_AP) {
+	if (changed & BSS_CHANGED_AP_PS) {
 		ar->ap_ps_enabled = info->ap_ps_enable;
 		ath12k_mac_ap_ps_recalc(ar);
 	}
